ner4opt/ner4opt.py

```diff
@@ -0,0 +1,135 @@
+import os
+import pathlib
+from typing import List
+
+import joblib
+
+from .constants import Constants
+from .exceptions import InvalidModelError
+from .features import Featurizer
+from .utils import format_entities, generate_feature_dictionary, l2_augment_sentence, load_torch_model
+
+
+class Ner4Opt(object):
+    """Extract named entities for a given optimization problem.
+
+    The main objective of the Ner4Opt library is to generate named entities for optimization problems. There are six
+    entities the current library extracts; CONST_DIR (constraint direction), LIMIT (limit),
+    OBJ_DIR (objective direction), OBJ_NAME (objective name), PARAM (parameter) and VAR (variable).
+
+    This library provides access to four main methods for extracting these entities.
+    A `lexical` model built using lexical features alone, a `lexical_plus` model which in addition to lexical features
+    uses optimization scenario specific features, a `semantic` model which uses a roberta based transformers model and
+    a hybrid model which combines all the best features of both lexical and semantic models to achieve best results on
+    our datasets.
+
+    Attributes
+    -------
+    model_name (str) : Specifies the model to use for extracting the entities.
+                       Options include `lexical`, `lexical_plus`, `semantic` and `hybrid`
+                       Default is `hybrid`
+    use_gpu (bool) :   Specifies the model to use gpu while calculating transformer based features.
+                       Default is False
+
+    Methods
+    -------
+    get_entities()
+
+    Attributes
+    -------
+    text (str) : Sample optimization problem
+
+    Returns named entities of a given optimization problem. A list of dictionaries for each entity
```

Comment:
should this have Returns______ line?

Reply:
is it for each "entity" OR for each "word" ??

Reply:
I added a returns line. I have looked at different entity output formats last few days. I have to realize that 'O' tags are not useful and generating starting character index , ending character index, score etc for 'O' is wasteful. So, just like standard spacy, huggingface outputs, our output will have "entity phrases" detected and for each we provide, starting/ending character index and a score.

Reply:
I like that we are back to standard inputs. Make sure that it is clear in the documentation, pydoc (that we only refer tags for tokens that are predicted to be an entity and omit the rest